crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,363 @@
+
+#[derive(Default, Debug, Serialize)]
+pub struct ExpresscheckoutRefundRequest {
+    unique_request_id: String,
+    amount: String,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for ExpresscheckoutRefundRequest {
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        let refund_req = Self {
+            unique_request_id: item.request.refund_id.clone(),
+            amount: item.request.amount.to_string(),
+        };
+        Ok(refund_req)
+    }
+}
+
+// Type definition for Refund Response
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone)]
+pub enum RefundStatus {
+    #[serde(rename = "ERROR")]
+    #[default]
+    Error,
+    #[serde(rename = "BAD REQUEST")]
+    BadRequest,
+    #[serde(rename = "NOT_FOUND")]
+    NotFound,
+    #[serde(rename = "SUCCESS")]
+    Success,
+    #[serde(rename = "PENDING")]
+    Pending,
+}
+
+impl From<RefundStatus> for enums::RefundStatus {
+    fn from(item: RefundStatus) -> Self {
+        match item {
+            RefundStatus::Success => Self::Success,
+            RefundStatus::Pending => Self::Pending,
+            _ => Self::Failure,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RefundResponse {
+    refunds: Vec<RefundStatusBlock>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RefundStatusBlock {
+    #[serde(deserialize_with = "deserialize_refund_status_default")]
+    status: RefundStatus,
+    id: Option<String>,
+    unique_request_id: String,
+}
+
+fn deserialize_refund_status_default<'de, D, RefundStatus>(
+    deserializer: D,
+) -> Result<RefundStatus, D::Error>
+where
+    RefundStatus: Default + Deserialize<'de>,
+    D: Deserializer<'de>,
+{
+    let opt = RefundStatus::deserialize(deserializer);
+    match opt {
+        Ok(v) => Ok(v),
+        _ => Ok(RefundStatus::default()),
+    }
+}
+
+fn get_status_from_refund_response(
+    refund_id: &str,
+    response: RefundResponse,
+) -> enums::RefundStatus {
+    response
+        .refunds
+        .into_iter()
+        .fold(enums::RefundStatus::Pending, |acc, r| {
+            if r.unique_request_id == refund_id {
+                enums::RefundStatus::from(r.status)
+            } else {
+                acc
+            }
+        })
+}
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, RefundResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        let refund_id = item.data.request.refund_id.clone();
+        let status = get_status_from_refund_response(&refund_id, item.response);
+        Ok(Self {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: item.data.request.refund_id.clone(),
```

Comment:
`item.data.request.refund_id.clone()` duplicate code. line no 330 you are doing this operation already.